src/pTSL.py

```diff
@@ -76,27 +113,97 @@ def val_corpus_helper(self, corpus_file):
             input_strings.append(' '.join(map(str, string)))
         return dict(zip(input_strings, cond_probs))
 
+    def eval_conditional_probabilities(self, corpus_file,
+                                       verbose=False,
+                                       train=False,
+                                       corpus_filename=None,
+                                       projection_filename=None):
+        '''
+            evaluate conditional probabilities given corpus data.
+
+            Input:
+                corpus_file: path to the corpus file.
+                verbose: output individual projections.
+                train: do not output conditional probabilities of train.
+                corpus_filename: path to save corpus conditional probabilities.
+                projection_filename: path to save individual projections.
+        '''
+        corpus_data = self.read_corpus_file(corpus_file)
+        corpus_projections = []
+        for data in corpus_data:
+            corpus_projections.append([data[0], self.get_projections(data[0])])
+        # currently designed to create csv and print separately, needing two separate iterations of projections

```

Comment:
I think this is fine for now; can be fixed later if people are running into efficiency issues.